### PR TITLE
feat(spacing tokens): add a codemod to transform current tokens into …

### PR DESCRIPTION
**Pull request by anitavincent**
…new spacing tokens

#### What is the purpose of this pull request?
Add a codemod for easier token migration

#### What problem is this solving?
token migratior

#### How should this be manually tested?
yarn workspace @vtex/admin-ui-codemod jest

or for more use cases
https://astexplorer.net/#/gist/f0c5c2b2fe8c0da2c625ff23b0012a06/124dda3bde830af5e5bc03c6bf21c8478bf7d2af

#### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Quality improvement (tests or refactors)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Trivial change (small fix or feature that doesn't impact functionalities)
- [ ] Requires change to documentation, which has been updated accordingly

### CODE REVIEW

File: packages/admin-ui-codemod/src/__tests__/space-tokens-review.js [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    "<div csx={{margin: '2rem 4rem', paddingBottom: '$xs', paddingRight: '$space-1'}} />",
+    "<div csx={{margin: '2rem 4rem', paddingBottom: '$space-0', paddingRight: '$space-1'}} />",
+    'change only deprecated space tokens'
+  )
+
+  defineInlineTest(
+    spaceTokenTransform,
+    {},
+    'export const baseline = style({ padding: true ? 1 : 2 })',
+    'export const baseline = style({ padding: true ? 1 : 2 })',
```
Comment (matheusps) [inline packages/admin-ui-codemod/src/__tests__/space-tokens-review.js:33]: Shouldn't it change?

File: packages/admin-ui-codemod/src/space-tokens-review.js [hunk 2]
```diff
@@ -0,0 +1,139 @@
+const spacingProps = {
```
Comment (lucasaarcoverde) [inline packages/admin-ui-codemod/src/space-tokens-review.js:1]: what about the first tokens that we have? (`1,2,3,4,5,6,7,8, px, 2px`) Shouldn't we replace them too?

File: packages/admin-ui-codemod/src/space-tokens-review.js [hunk 3]
```diff
@@ -79,13 +92,17 @@ function transformValue(j, propKey, value) {
     })
   }
 
-  if (typeof value.value !== 'string') {
+  // if (typeof value.value === 'integer') {
+
+  // }
```
Comment (lucasaarcoverde) [inline packages/admin-ui-codemod/src/space-tokens-review.js:97]: This can be removed

File: packages/admin-ui-codemod/src/space-tokens-review.js [hunk 4]
```diff
@@ -33,22 +33,35 @@ const spacingProps = {
...
+  if (typeof value.alternate.value === 'string' || typeof value.alternate.value === 'number') {
     value.alternate = j.stringLiteral(transform(value.alternate.value))
   }
 }
 
 function replaceNegative(j, key, callExp) {
   const tokenVal = callExp.arguments[0].value
 
-  if (callExp.callee.name !== 'negative' || typeof tokenVal !== 'string') {
+  if (callExp.callee.name !== 'negative' || (typeof tokenVal !== 'string' && typeof tokenVal !== 'number')) {
```
Comment (lucasaarcoverde) [inline packages/admin-ui-codemod/src/space-tokens-review.js:64]:
```suggestion
function isStringOrNumber(value) {
  return typeof value === 'string' || typeof value === 'number'
}

if (isStringOrNumber(value.consequent.value)) {
...

if (isStringOrNumber(value.alternate.value)) {
...

  if (callExp.callee.name !== 'negative' || !isStringOrNumber(tokenVal)) {
  ...
```

File: packages/admin-ui-codemod/src/space-tokens-review.js [hunk 5]
```diff
@@ -33,22 +33,35 @@ const spacingProps = {
...
+  0: '$space-0',
+  1: '$space-1',
+  2: '$space-2',
+  3: '$space-3',
+  4: '$space-4',
+  5: '$space-5',
+  6: '$space-6',
+  7: '$space-7',
+  8: '$space-8',
+  '2px': '$space-05',
```
Comment (lucasaarcoverde) [inline packages/admin-ui-codemod/src/space-tokens-review.js:46]: `px` needs a conversion as well, otherwise it will be an invalid value

**Review by matheusps — APPROVED** [review]
LGTM

File: packages/admin-ui-codemod/src/__tests__/space-tokens-review.js [hunk 6]
```diff
@@ -0,0 +1,270 @@
+const defineInlineTest = require('jscodeshift/dist/testUtils').defineInlineTest
```
Comment (mairatma) [inline packages/admin-ui-codemod/src/__tests__/space-tokens-review.js:1]: I really like these tests, @anitavincent ! They're pretty straight forward to read and have good coverage! 👏

**Review by mairatma — APPROVED** [review]

**Pull request merged**